v3/metadata/metadata.go

```diff
@@ -66,26 +64,13 @@ func Get(ctx context.Context, endpoint Endpoint) (string, error) {
 // Important note: Run this code as privileged user.
 // Not Windows compatible.
 func FromCdRom(endpoint Endpoint) (string, error) {
-	const target = "/tmp/cloud-init-mount"
-
-	_, err := os.Stat(CdRomPath)
-	if errors.Is(err, os.ErrNotExist) {
-		return "", fmt.Errorf("CD-ROM(iso9660) %s, not found: %w", CdRomPath, err)
-	}
+	iso, err := diskfs.Open(CdRomPath, diskfs.WithOpenMode(diskfs.ReadOnly))
 	if err != nil {
-		return "", fmt.Errorf("stat %s: %w", CdRomPath, err)
-	}
-
-	if err := os.MkdirAll(target, os.ModePerm); err != nil {
-		return "", fmt.Errorf("create mountpoint directory: %w", err)
-	}
-
-	if err := unix.Mount(CdRomPath, target, "iso9660", unix.MS_RDONLY, ""); err != nil {
-		return "", fmt.Errorf("mount %s: %w", CdRomPath, err)
+		return "", fmt.Errorf("disk open: %w", err)
 	}
-	defer unix.Unmount(target, 0) //nolint: errcheck
+	defer iso.File.Close()
 
-	return getFileMetaDataValue(filepath.Join(target, "meta-data"), string(endpoint))
+	return getFileMetaDataValue(iso.File, string(endpoint))
```

Comment:
If the only thing you do is just to reuse the underlying file descriptor to read it, can't you just `os.Open()` directly without the dependency on diskfs ?  

Unless I missed something that the lib does

Reply:
I will try to use and make work:
 using `Disk.GetFilesystem` then `FileSystem.OpenFile` ?
Otherwise, yes if it's not working I can use open directly